### PR TITLE
Search HTTP Request Content: Enables pinpointing requests with specific data

### PR DESCRIPTION
**Pull request by abuzer**
*Description:*
I'm thrilled to propose a new feature for Laravel Telescope: the ability to search the content of HTTP requests. This feature is aimed at enhancing the debugging capabilities of Telescope by allowing developers to locate specific requests based on their content.

*Key Highlights:*
Search HTTP Request Content: Enables pinpointing requests with specific data.
Seamless UI Integration: Maintains the intuitive experience of Telescope.
Optimized for Performance: Designed to be efficient and minimally impactful on performance.

*Why This Feature?*
As a Laravel developer, I've often faced the challenge of finding specific requests among many. This feature was born out of the necessity to streamline this process, making our lives a bit easier and our debugging faster.

*My Journey & Request for Guidance:*
This is my first attempt at contributing to an open-source project. I am fully aware that there may be room for improvement and I am eager to learn from the community. Your feedback will not only help enhance this feature but also aid in my growth as a developer. I am open to suggestions and committed to making the necessary adjustments to meet the project's standards.

*Closing Thoughts:*
I believe this feature will make Laravel Telescope an even more powerful tool for developers. I'm looking forward to your constructive feedback and hope to contribute positively to this amazing community.

Thank you for considering my contribution.

image_2024_03_27T08_16_56_802Z
image_2024_03_27T08_17_01_763Z

### CODE REVIEW

**Comment by driesvints** [comment]
Thanks @abuzer. Can you undo some of the styling changes in this PR so we can focus on the actual changes involved?

**Comment by abuzer** [comment]
@driesvints There is not style changes, just a text field added? Are you referring to it?

File: src/Storage/EntryModel.php [hunk 1]
```diff
@@ -65,11 +65,12 @@ class EntryModel extends Model
...
-                ->whereTag($query, $options)
-                ->whereFamilyHash($query, $options)
-                ->whereBeforeSequence($query, $options)
-                ->filter($query, $options);
+            ->whereBatchId($query, $options)
+            ->whereTag($query, $options)
+            ->whereContent($query, $options)
+            ->whereFamilyHash($query, $options)
+            ->whereBeforeSequence($query, $options)
+            ->filter($query, $options);
```
Comment (driesvints) [inline src/Storage/EntryModel.php]: This.

File: src/Storage/EntryModel.php [hunk 2]
```diff
@@ -116,7 +117,7 @@ protected function whereBatchId($query, EntryQueryOptions $options)
     protected function whereTag($query, EntryQueryOptions $options)
     {
         $query->when($options->tag, function ($query, $tag) {
-            $tags = collect(explode(',', $tag))->map(fn ($tag) => trim($tag));
+            $tags = collect(explode(',', $tag))->map(fn($tag) => trim($tag));
```
Comment (driesvints) [inline src/Storage/EntryModel.php]: This.

File: src/Storage/EntryQueryOptions.php [hunk 3]
```diff
@@ -57,12 +64,14 @@ class EntryQueryOptions
...
-                ->tag($request->tag)
-                ->familyHash($request->family_hash)
-                ->limit($request->take ?? 50);
+            ->batchId($request->batch_id)
+            ->uuids($request->uuids)
+            ->beforeSequence($request->before)
+            ->tag($request->tag)
+            ->familyHash($request->family_hash)
+            ->limit($request->take ?? 50)
+            ->content($request->content);
```
Comment (driesvints) [inline src/Storage/EntryQueryOptions.php]: This.

**Comment by abuzer** [comment]
Updated.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by abuzer** [comment]
@taylorotwell This is a great package. Plus I have checked this has been PR-ed multiple time. So there is a need of it. 
There is no much code involved in it, but we have had multiple time need for this search feature to filter down the request. 

Lets say a request was logged 8 hours ago. For now we have to scroll down to almost 20 pages to find that request. Which is not efficient. Plus this function only works when there is a search query. Else the original code keeps on working as is. 

Can you please reconsider it?

**Comment by driesvints** [comment]
@abuzer seems for now we have no plans for this, sorry. You can always fork Horizon if you want.

**Pull request closed without merging**